contrib/epee/include/net/net_helper.h

```diff
@@ -155,97 +159,10 @@ namespace net_utils
       return connect(addr, std::to_string(port), timeout);
     }
 
-    inline
-			try_connect_result_t try_connect(const std::string& addr, const std::string& port, std::chrono::milliseconds timeout)
-		{
-				m_deadline.expires_from_now(timeout);
-				boost::unique_future<boost::asio::ip::tcp::socket> connection = m_connector(addr, port, m_deadline);
-				for (;;)
-				{
-					m_io_service.reset();
-					m_io_service.run_one();
 
-					if (connection.is_ready())
-						break;
-				}
-
-				m_ssl_socket->next_layer() = connection.get();
-				m_deadline.cancel();
-				if (m_ssl_socket->next_layer().is_open())
-				{
-					m_connected = true;
-					m_deadline.expires_at(std::chrono::steady_clock::time_point::max());
-					// SSL Options
-					if (m_ssl_options.support == epee::net_utils::ssl_support_t::e_ssl_support_enabled || m_ssl_options.support == epee::net_utils::ssl_support_t::e_ssl_support_autodetect)
-					{
-						if (!m_ssl_options.handshake(*m_ssl_socket, boost::asio::ssl::stream_base::client, addr, timeout))
-						{
-							if (m_ssl_options.support == epee::net_utils::ssl_support_t::e_ssl_support_autodetect)
-							{
-								boost::system::error_code ignored_ec;
-								m_ssl_socket->next_layer().shutdown(boost::asio::ip::tcp::socket::shutdown_both, ignored_ec);
-								m_ssl_socket->next_layer().close();
-								m_connected = false;
-								return CONNECT_NO_SSL;
-							}
-							else
-							{
-								MWARNING("Failed to establish SSL connection");
-								m_connected = false;
-								return CONNECT_FAILURE;
-							}
-						}
-					}
-					return CONNECT_SUCCESS;
-				}else
-				{
-					MWARNING("Some problems at connect, expected open socket");
-					return CONNECT_FAILURE;
-				}
 
-		}
```

Comment:
That seems to do a lot more than exposing last error.

Reply:
Originally this changed a bunch of lines for DANE/TLSA. This can be rolled back somewhat for this more limited PR, but there are still a number of changes for `boost::system::error_code` return from `ssl_options_t::handshake`.

There was some redundant checks that got removed, and it seemed like an opportunity to just move a chunk of code to the cpp. The amount moved wasn't huge in the end.

Reply:
Maybe it hasn't changed, but it makes reviewing a lot more annoying since manually reading two bits of source to see they're indeed the same is a pita. But hey, they're not *that* large I guess,

Reply:
I reverted the move to the cpp, to make it easier to read.